catalog/MDCCatalog/MDCCatalogDebugAlert.swift

```diff
@@ -0,0 +1,200 @@
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import UIKit
+
+import MaterialComponents.MaterialDialogs
+
+class MDCCatalogDebugSetting {
+  init(title: String) {
+    self.title = title
+  }
+
+  public var title: String
+  public var setter: (Bool) -> Void = { _ in }
+  public var getter: () -> Bool = { false }
+}
+
+class MDCCatalogDebugAlert: UICollectionViewController {
+  fileprivate var settings = [MDCCatalogDebugSetting]()
+
+  fileprivate let layout = UICollectionViewFlowLayout()
+  fileprivate let dialogTransitionController = MDCDialogTransitionController()
+
+  fileprivate var numberOfRows: Int {
+    return settings.count + 1
+  }
+  fileprivate var dismissCellRow: Int {
+    return settings.count
+  }
+
+  override var preferredContentSize: CGSize {
+    get { return CGSize.init(width: 400, height: self.numberOfRows * 44) }
+    set { super.preferredContentSize = newValue }
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  init(settings: [MDCCatalogDebugSetting]) {
+    self.settings = settings
+
+    layout.minimumLineSpacing = 0
+    layout.minimumInteritemSpacing = 0
+
+    super.init(collectionViewLayout: layout)
+
+    self.transitioningDelegate = dialogTransitionController
+    self.modalPresentationStyle = .custom
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    self.collectionView?.backgroundColor = UIColor.white
+    self.collectionView?.register(MDCCatalogDebugToggleCell.self,
+                                  forCellWithReuseIdentifier: "toggle")
+    self.collectionView?.register(MDCCatalogDebugDismissCell.self,
+                                  forCellWithReuseIdentifier: "dismiss")
+  }
+
+  override func viewWillLayoutSubviews() {
+    super.viewWillLayoutSubviews()
+
+    layout.itemSize = CGSize.init(width: self.view.frame.width, height: 44)
+  }
+
+  override func collectionView(_ collectionView: UICollectionView,
+                               cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+    if indexPath.row == self.dismissCellRow {
+      return collectionView.dequeueReusableCell(withReuseIdentifier: "dismiss", for: indexPath)
+    }
+    let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "toggle", for: indexPath)
+    if let cell = cell as? MDCCatalogDebugToggleCell {
+      cell.setting = settings[indexPath.row]
+    }
+    return cell
+  }
+
+  override func collectionView(_ collectionView: UICollectionView,
+                               numberOfItemsInSection section: Int) -> Int {
+    return self.numberOfRows
```

Comment:
Try removing all `self.`s and only add back in the ones Xcode forces you to.

Reply:
DOne.